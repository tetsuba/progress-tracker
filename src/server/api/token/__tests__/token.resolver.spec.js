const { createTestServer } = require('../../../test/serverTestHelper')
const {
  CONFIRM_TOKEN_QUERY,
} = require('../../../../client/api/token/token.query')
const {
  tokenMockSuccess,
  tokenMockError,
} = require('../../../../test/mockApi/token/tokenMockData')

describe('token.resolver', () => {
  describe('@Query', () => {
    describe('confirmToken', () => {
      test('should return token confirmed', async () => {
        const ctx = {
          user: { id: 'userId1234' },
          models: {
            Token: {
              findOne: jest.fn(() => tokenMockSuccess),
            },
          },
        }
        const { query } = createTestServer(ctx)
        const res = await query({
          query: CONFIRM_TOKEN_QUERY,
          variables: { token: 'token' },
        })
        expect(res).toMatchSnapshot()
      })

      test('should return an error token expired', async () => {
        const ctx = {
          user: { id: 'userId1234' },
          models: {
            Token: {
              findOne: jest.fn(() => tokenMockError),
            },
          },
        }
        const { query } = createTestServer(ctx)
        const res = await query({
          query: CONFIRM_TOKEN_QUERY,
          variables: { token: 'token' },
        })
        expect(res).toMatchSnapshot()
      })
    })
  })
})
