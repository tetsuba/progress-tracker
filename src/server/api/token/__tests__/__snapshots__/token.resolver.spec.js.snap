// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`token.resolver @Query confirmToken should return an error token expired 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: TOKEN_EXPIRED],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`token.resolver @Query confirmToken should return token confirmed 1`] = `
Object {
  "data": Object {
    "confirmToken": Object {
      "success": "token confirmed",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
